pre-sync/oci-image-verification/README.md

```diff
@@ -0,0 +1,156 @@
+# Sample OCI Image Signature Verification with Config Sync
+
+This sample demonstrates how to verify the signature of an OCI image that Config
+Sync is managing. The webhook monitors the `configsync.gke.io/image-to-sync`
+annotation on specified RootSync or RepoSync resources, which contains the full
```

Comment:
> The webhook monitors the `configsync.gke.io/image-to-sync`
> annotation on specified RootSync or RepoSync resources, which contains the full
> URL of the image to sync.

This is not technically correct. The webhook doesn't monitor anything. It's the k8s-api-server that calls the webhook server for validation.

I would suggest something like below:

This sample demonstrates how to verify the signature of an OCI image that Config
Sync is managing using [Cosign](https://github.com/sigstore/cosign). It leverages a ValidatingWebhookConfiguration object, along with a validating webhook server to intercept update requests for RootSync and RepoSync objects. Config Sync updates the `configsync.gke.io/image-to-sync` annotation of RootSync and RepoSync objects after it fetches a new image digest successfully. The validating webhook server compares the values between the old annotation and the new annotation, and runs the validation when a change is detected.